tests/framework/artifact/test_hdf.py

```diff
@@ -306,3 +306,32 @@ def test_EntityKey_with_name():
     assert key.group == f'/{type_}/{name}'
     assert key.path == f'/{type_}/{name}/{measure}'
     assert key.with_measure('prevalence') == hdf.EntityKey(f'{type_}.{name}.prevalence')
+
+
+def test_entity_key_equality():
+
+    type_ = 'cause'
+    name = 'diarrheal_diseases'
+    measure = 'incidence'
+    string = f'{type_}.{name}.{measure}'
+    key = hdf.EntityKey(string)
+
+    class NonString:
+        def __str__(self):
+            return string
+
+    nonstring = NonString()
+
+    assert key == string, 'Comparision using __eq__ between string object and equivalent EntityKey failed'
+    assert not (key != string), 'Comparision using __ne__ between string object and equivalent EntityKey failed'
```

Comment:
Does this fail unless you define __ne__?  I though python would infer it from __eq__.

Reply:
python does **not** infer it from eq. Something I learned today.

Reply:
You an me both.  Hmm.